src/components/TeacherViews/Onboarding/SchoolSelection/SchoolSelection.view.tsx

```diff
@@ -0,0 +1,82 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useCallback, useEffect } from "react";
+import { Control, UseFormTrigger, useForm } from "react-hook-form";
+
+import useSchoolPicker from "@/components/TeacherComponents/ResourcePageSchoolPicker/useSchoolPicker";
+import ResourcePageSchoolPicker from "@/components/TeacherComponents/ResourcePageSchoolPicker";
+import OnboardingForm from "@/components/TeacherComponents/OnboardingForm/OnboardingForm";
+import {
+  OnboardingFormProps,
+  SchoolSelectFormProps,
+  schoolSelectFormSchema,
+} from "@/components/TeacherComponents/OnboardingForm/OnboardingForm.schema";
+
+export const SchoolSelectionView = () => {
+  const { formState, setValue, handleSubmit, control, trigger } =
+    useForm<SchoolSelectFormProps>({
+      resolver: zodResolver(schoolSelectFormSchema),
+      mode: "onBlur",
+      defaultValues: {
+        newsletterSignUp: true,
+      },
+    });
+
+  const setSchoolDetailsInForm = useCallback(
+    (value: string, name: string) => {
+      setValue("school", value, {
+        shouldValidate: true,
+      });
+      setValue("schoolName", name, {
+        shouldValidate: true,
+      });
+    },
+    [setValue],
+  );
+
+  const {
+    selectedSchool,
+    setSelectedSchool,
+    schoolPickerInputValue,
+    setSchoolPickerInputValue,
+    schools,
+  } = useSchoolPicker({ withHomeschool: false });
+
+  useEffect(() => {
+    if (selectedSchool && schoolPickerInputValue !== "") {
+      setSchoolDetailsInForm(selectedSchool.toString(), schoolPickerInputValue);
+    }
+  }, [selectedSchool, schoolPickerInputValue, setSchoolDetailsInForm]);
+
+  const onSchoolPickerInputChange = (value: React.SetStateAction<string>) => {
+    if (value === "") {
+      setSchoolDetailsInForm("", "");
+    }
+    setSchoolPickerInputValue(value);
+  };
+
+  return (
+    <OnboardingForm
+      control={control as Control<OnboardingFormProps>}
+      trigger={trigger as UseFormTrigger<OnboardingFormProps>}
+      formState={formState}
+      heading="Select your school"
+      handleSubmit={handleSubmit}
+      continueDisabled={
+        formState.errors?.school !== undefined || !formState.isValid
```

Comment:
This is just for me because i'm new to `react-hook-form`, but what does this line actually do? What does `formState.errors?.school !== undefined` do that  `!formState.isValid` doesn't?

Also the negation is really hard on my brain. Could it be flipped for easier comprehension?

```typescript
const canSubmit = formState.errors?.school === undefined && formState.isValid
...
continueDisabled={!canSubmit}
```

Reply:
hmm good question, so `isValid` will be false from the start as the required fields are not filled out, and I think it will also be false if there are errors set, but the errors will only be set at certain times eg when attempting to continue when the form is invalid - so I'm not sure we do need the check for errors here!